packages/react/src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { ReactElement } from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '../IconButton';
+import Select from '../Select';
+
+export interface paginationProps {
+  pageSize: number;
+  totalPages: number;
+  currentPage: number;
+  totalItems: number;
+  currentItemCount: number;
+  hasPrevious: boolean;
+  hasNext: boolean;
+  mobileView?: boolean;
+  handlePageSizeChange: (pagesize: string) => void;
+  handlePageChange: (pagenumber: number) => void;
+}
+
+const Pagination = (props: paginationProps) => {
+  const mobileView = props.mobileView ? props.mobileView : false;
+
+  const previousPage = props.currentPage - 1;
+  const startIndex =
+    props.currentPage === 1 ? 1 : props.pageSize * previousPage + 1;
+  const endIndex = startIndex + props.currentItemCount - 1;
+  const current_page_index = (
+    <p className={mobileView ? 'Page_Index_Sm' : ''}>
+      Showing {startIndex}-{endIndex} of {props.totalItems}
+    </p>
+  );
+  const changePage = (e: React.MouseEvent, pagenumber: number) => {
+    props.handlePageChange(pagenumber);
+  };
+
+  return (
+    <div className={`${mobileView ? 'Pagination_Sm' : ''} Pagination`}>
+      <div
+        className={`${
+          mobileView ? 'Page_Size_Selector_Sm' : ''
+        } Page_Size_Selector`}
+      >
+        <Select
+          value={props.pageSize}
+          className={'Page_Size_Options'}
+          placeholder="Select page size"
+          label={'Show'}
+          onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
+            props.handlePageSizeChange(e.target.value)
+          }
+          options={[
+            {
+              key: '10',
+              value: '10',
+              label: '10 records'
+            },
+            {
+              key: '50',
+              value: '50',
+              label: '50 records'
+            },
+            {
+              key: '100',
+              value: '100',
+              label: '100 records'
+            }
+          ]}
+        />
+      </div>
+      <nav
+        id="page_index"
+        className={'Page_Index_Wrapper'}
```

Comment:
I'm not sure this class name is needed, we should be able to target uniquely in css with the following selector: `.Pagination nav`.